libs/java/dal/db/dal-db/src/main/java/ca/bc/gov/tno/dal/db/services/ContentService.java

```diff
@@ -346,50 +366,105 @@ public Content update(Content entity) {
+        });
+
     if (entity.getContentActions().size() > 0) {
       entity.getContentActions().forEach((action) -> {
         if (action.getActionId() == 0)
           contentActionService.add(action);
-        else
+        else if (action.getUpdatedOn() == null)
           contentActionService.update(action);
       });
     }
+    actions.stream()
+        .filter((a) -> !entity.getContentActions().stream().anyMatch((ca) -> ca.getActionId() == a.getActionId()))
+        .forEach((a) -> {
+          contentActionService.delete(a);
+        });
+
     if (entity.getContentTags().size() > 0) {
       entity.getContentTags().forEach((tag) -> {
-        if (tag.getCreatedOn() == null) // TODO: I don't think this will work presently
+        if (tag.getCreatedOn() == null)
           contentTagService.add(tag);
-        else
+        else if (tag.getUpdatedOn() == null)
           contentTagService.update(tag);
       });
     }
+    tags.stream()
+        .filter((t) -> !entity.getContentTags().stream().anyMatch((ct) -> ct.getTagId() == t.getTagId()))
+        .forEach((t) -> {
+          contentTagService.delete(t);
+        });
+
     if (entity.getContentCategories().size() > 0) {
       entity.getContentCategories().forEach((category) -> {
         if (category.getCategoryId() == 0)
           contentCategoryService.add(category);
-        else
+        else if (category.getUpdatedOn() == null)
           contentCategoryService.update(category);
       });
     }
+    categories.stream()
+        .filter(
+            (c) -> !entity.getContentCategories().stream().anyMatch((cc) -> cc.getCategoryId() == c.getCategoryId()))
+        .forEach((c) -> {
+          contentCategoryService.delete(c);
+        });
+
     if (entity.getContentTonePools().size() > 0) {
       entity.getContentTonePools().forEach((tone) -> {
         if (tone.getTonePoolId() == 0)
           contentToneService.add(tone);
-        else
+        else if (tone.getUpdatedOn() == null)
           contentToneService.update(tone);
       });
     }
+    tonePools.stream()
+        .filter((t) -> !entity.getContentTonePools().stream().anyMatch((tp) -> tp.getTonePoolId() == t.getTonePoolId()))
+        .forEach((t) -> {
+          contentToneService.delete(t);
+        });
+
     if (entity.getTimeTrackings().size() > 0) {
       entity.getTimeTrackings().forEach((time) -> {
-        if (time.getCreatedOn() == null) // TODO: I don't think this will work presently
+        if (time.getCreatedOn() == null)
           timeTrackingService.add(time);
-        else
+        else if (time.getUpdatedOn() == null)
           timeTrackingService.update(time);
       });
     }
+    // TODO: Need to test this because java is horrible and may not
+    // serialize/deserialize dates consistently.
+    timeTrackings.stream()
+        .filter((t) -> !entity.getTimeTrackings().stream()
+            .anyMatch((tt) -> tt.getUserId() == t.getUserId() && tt.getCreatedOn() == t.getCreatedOn()))
```

Comment:
Only delete time tracking that doesn't have a matching date+user.  I worry that this may result in deleting them all, but will need to test with frontend.